src/utils/utils_posix_math.c

```diff
@@ -0,0 +1,20 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "utils_math.h"
+
+// Retrieves the position of the leftmost set bit.
+// The position of the bit is counted from 0
+// e.g. for 01000011110 the position equals 9.
+size_t getLeftmostSetBitPos(size_t num) {
+    assert(num != 0 &&
+           "Finding leftmost set bit when number equals zero is undefined");
+    // From C++20 countl_zero could be used for that.
+    return 31 - __builtin_clz(num);
```

Comment:
Are you sure that `size_t` will always be 32 bit? Also, see that there are clzl and clzll variants of this builtin, too - pick one suitable for size_t type.

Reply:
You're right. I changed it to use `sizeof` operator.